### PR TITLE
[2.6] Add NGINX 1.0.4

### PR DESCRIPTION
**Pull request by pennyscissors**
Issue: https://github.com/rancher/rancher/issues/35128
Related PR: https://github.com/rancher/ingress-nginx/pull/67

### CODE REVIEW

**Review by nickgerace — DISMISSED** [review]

File: rke/templates/nginx-v0.49.3.go [hunk 1]
```diff
@@ -33,13 +33,18 @@ metadata:
...
     app.kubernetes.io/name: ingress-nginx
     app.kubernetes.io/instance: ingress-nginx
-    app.kubernetes.io/version: 0.48.1
+    app.kubernetes.io/version: 0.49.3
     app.kubernetes.io/component: controller
   name: nginx-configuration
   namespace: ingress-nginx
-{{- if .Options }}
 data:
+{{- if $v := (index .Options "allow-snippet-annotations") }}
```
Comment (kinarashah) [inline rke/templates/nginx-v0.49.3.go]: We should add # Rancher specific change comment here

File: rke/templates/nginx-v0.49.3.go [hunk 2]
```diff
@@ -620,6 +625,8 @@ spec:
                   fieldPath: metadata.namespace
       restartPolicy: OnFailure
       serviceAccountName: ingress-nginx-admission
+      nodeSelector:
```
Comment (kinarashah) [inline rke/templates/nginx-v0.49.3.go]: Is this coming from upstream? nodeSelector already exists above for `ingress-nginx-admission-create` L605, so don't need to add it here.

File: rke/templates/nginx-v0.49.3.go [hunk 3]
```diff
@@ -673,6 +680,8 @@ spec:
...
```
Comment (kinarashah) [inline rke/templates/nginx-v0.49.3.go]: Same as above, nodeSelector already existing on spec: L658.

File: rke/templates/templates.go [hunk 4]
```diff
@@ -164,8 +165,12 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			// New ingress template introduced for 1.15.12-rancher1-1, 1.16.10-rancher1-1, 1.17.6-rancher1-1
 			">=1.15.12-rancher1-1 <1.16.1-rancher1-1": nginxIngressV11512,
 			">=1.16.10-rancher1-1 <1.17.0-rancher1-1": nginxIngressV11512,
-			">=1.17.6-rancher1-1 <1.21.0-rancher1-1":  nginxIngressV11512,
-			">=1.21.0-rancher1-1":                     nginxIngressv0481,
+			">=1.17.6-rancher1-1 <1.19.15-rancher1-2": nginxIngressV11512,
+			">=1.19.15-rancher1-3 <1.20.0-rancher0":   nginxIngressv0493,
+			">=1.20.0-rancher0 <1.20.11-rancher1-2":   nginxIngressV11512,
+			">=1.20.11-rancher1-3 <1.21.0-rancher0":   nginxIngressv0493,
+			">=1.21.0-rancher0 <1.21.5-rancher1-1":    nginxIngressv0481,
```
Comment (kinarashah) [inline rke/templates/templates.go]: This should be `<1.21.5-rancher1-2`. Current way would mean 1.21.5-rancher1-1 has no template. 
Also, check if our `go generate` caught and logged this as a warning. If it didn't, could be a bug.

Comment (pennyscissors) [reply rke/templates/templates.go]: It did come up, but I missed it. Thanks for pointing it out.

File: rke/k8s_version_info.go [hunk 5]
```diff
@@ -510,6 +522,18 @@ func loadK8sVersionInfo() map[string]v3.K8sVersionInfo {
 			MinRancherVersion: "2.5.6-rc0",
 			MinRKEVersion:     "1.2.0-rc0",
 		},
+		"v1.20.11-rancher1-1": {
```
Comment (kinarashah) [inline rke/k8s_version_info.go:525]: Note: rancher1-1 and rancher1-2 aren't introduced here, just carrying the version forward from 2.5. Not a behavior change, all versions will load for 2.6 anyway. Adding them now to keep the versions consistent.

**Review by kinarashah — APPROVED** [review]

**Pull request merged**